File: bon-appetit.rb
def bonAppetit(bill, k, b)
    total = bill.inject { |sum, n| sum + n }
    split = (total - bill[k]) / 2 
    
    if split == b
        puts 'Bon Appetit'
    else
        b -= split
        puts b
    end
end

File: bon-appetit.js
function bonAppetit(bill, k, b) {
    let split = 0
    let amountAnna = b
    for (let i = 0; i < bill.length; i++) {
        if (i !== k) {
            split += bill[i]
        }
    }

    if (split / 2 === b) {
        console.log('Bon Appetit');
    } else {
        amountAnna -= split / 2;
        console.log(amountAnna);
    }
}